rskj-core/src/main/java/co/rsk/mine/BlockToMineBuilder.java

```diff
@@ -101,6 +102,24 @@ public BlockToMineBuilder(
         this.minerUtils = minerUtils;
     }
 
+    /**
+     * Creates a pending block based on the parent block header. Pending block is temporary, not connected to the chain and
+     * includes txs from the mempool.
+     *
+     * @param parentHeader block header a "pending" block is based on.
+     *
+     * @return "pending" block result.
+     * */
+    @Nonnull
+    public BlockResult buildPending(@Nonnull BlockHeader parentHeader) {
+        Coin minimumGasPrice = minimumGasPriceCalculator.calculate(parentHeader.getMinimumGasPrice());
+        List<BlockHeader> uncles = getUnclesHeaders(parentHeader);
+        List<Transaction> txs = getTransactions(new ArrayList<>(), parentHeader, minimumGasPrice);
+        Block newBlock = createBlock(Collections.singletonList(parentHeader), uncles, txs, minimumGasPrice, null);
+
```

Comment:
I think we have some duplication between this method and `co.rsk.mine.BlockToMineBuilder#build`. We could address it, WDYT?

Reply:
the thing is that these lines are a bit specific to each method, so I wouldn't extract common stuff into a separate method. Also methods themselves are not big

Reply:
If I get it right, the following 3 lines are conceptually performing the same operation: creating a pending block.
```
        Coin minimumGasPrice = minimumGasPriceCalculator.calculate(newBlockParentHeader.getMinimumGasPrice());
        final List<Transaction> txs = getTransactions(txsToRemove, newBlockParentHeader, minimumGasPrice);
        final Block newBlock = createBlock(mainchainHeaders, uncles, txs, minimumGasPrice, extraData);
```
IMO they could be made to a common `createPendingBlock` method without issues, just minor adaptations on variable creation. But not big deal, ofc.

Reply:
actually `buildPending()` creates a "pending" one that's used by json-rpc. Another one - `build()` - is used by minerServer for creating block templates. The second one has some side-effects, eg. removing txs from the mempool,  using list of `mainchainHeaders`, setting some extraData etc. That's why I said that they are a bit specific + small portion of semi-duplication (3 lines)